### PR TITLE
fix: Uncaught type error by `sticky-event`

### PR DESCRIPTION
**Pull request by mudana-grune**
## Redmine
task: [199485](https://redmine.weseek.co.jp/issues/119485) (If merged close this task.)

## Issue 
https://youtrack.weseek.co.jp/issue/GW-7946
- sticky-event Uncaught TypeError on open editor mode
- sticky-event Uncaught TypeError on some page transition

## Fix
- Create custom useSticky hook to observe sticky elements in use-sticky-utils
- Get isSticky status with useSticky hook
- Remove implementation of useSticky hook form react-use-sticky
- Remove isSticky state from Fab componnet
- Remove sticky-events and react-use-sticky library

## Next task
- Create Cypress test
https://youtrack.weseek.co.jp/issue/GW-7949

### CODE REVIEW

File: packages/app/src/components/Fab.tsx [hunk 1]
```diff
@@ -26,55 +26,51 @@ export const Fab = (): JSX.Element => {
...
   useRipple(createBtnRef, { rippleColor: 'rgba(255, 255, 255, 0.3)' });
 
+  // Get sticky status
+  const isSticky = useSticky('#grw-fav-sticky-trigger');
+
   /**
    * After the fade animation is finished, fix the button display status.
    * Prevents the fade animation occurred each time by button components rendered.
    * Check Fab.module.scss for fade animation time.
    */
```
Comment (jam411) [inline packages/app/src/components/Fab.tsx:42]: @mudana-grune 
Please move this comment to right place or update or delete.

File: packages/app/src/components/Fab.tsx [hunk 2]
```diff
@@ -26,55 +26,51 @@ export const Fab = (): JSX.Element => {
...
-      }
-    }, 500);
-    return () => clearTimeout(timer);
+    if (isSticky) {
+      setIsStickyApplied(true);
+    }
   }, [isSticky]);
 
-  const stickyChangeHandler = useCallback((event) => {
-    logger.debug('StickyEvents.CHANGE detected');
```
Comment (jam411) [inline packages/app/src/components/Fab.tsx:54]: logger はいらないかな

File: packages/app/src/components/Fab.tsx [hunk 3]
```diff
@@ -26,55 +26,51 @@ export const Fab = (): JSX.Element => {
...
-        setButtonClasses('');
-      }
-      else {
-        setAnimateClasses('invisible');
-      }
-    }, 500);
-    return () => clearTimeout(timer);
+    if (isSticky) {
+      setIsStickyApplied(true);
+    }
```
Comment (jam411) [inline packages/app/src/components/Fab.tsx:48]:
```
The button will showing at first load, then disappear. So I set isStickyApplied to true if isSticky is true at the first time.
If isStickyApplied is false, the effect will not run, button not appear at first load
```

https://wsgrowi.slack.com/archives/C029KNKD1RS/p1681352128540049?thread_ts=1681174479.582449&cid=C029KNKD1RS

**Review by jam411 — APPROVED** [review]

File: packages/app/src/stores/use-sticky.ts
Comment (yuki-takei) [inline packages/app/src/stores/use-sticky.ts:1]: src/client/side-effects に入れてください

**Review by yuki-takei — CHANGES_REQUESTED** [review]

**Review by jam411 — APPROVED** [review]

**Pull request merged**